auction-server/src/auction/service/verification.rs

```diff
@@ -533,82 +529,206 @@ impl Service<Svm> {
-                        &transaction,
+                        &bid_data.transaction,
                         &swap_instruction,
-                        self.config
-                            .chain_config
-                            .express_relay
-                            .mint_input_account_position_swap,
+                        svm_config
+                            .swap_instruction_account_positions
+                            .mint_output_account,
                     )
                     .await?;
-                let mint_output = self
+
+
+                let opp = self
+                    .opportunity_service
+                    .get_live_opportunity_by_id(GetOpportunityByIdInput {
+                        opportunity_id: bid_data.opportunity_id,
+                    })
+                    .await
+                    .ok_or(RestError::BadParameters(
+                        "No swap opportunity with the given id found".to_string(),
+                    ))?;
+
+                let opp_swap_data = match opp.program {
+                    Swap(opp_swap_data) => opp_swap_data,
+                    _ => {
+                        return Err(RestError::BadParameters(format!(
+                            "Opportunity with id {} is not a swap opportunity",
+                            bid_data.opportunity_id
+                        )));
+                    }
+                };
+                let (
+                    bid_amount,
+                    expected_input_token,
+                    expected_input_amount,
+                    expected_output_token,
+                    expected_output_amount,
+                ) = match opp_swap_data.quote_tokens.clone() {
+                    QuoteTokens::InputTokenSpecified {
+                        input_token,
+                        output_token,
+                        ..
+                    } => (
+                        swap_data.amount_output,
+                        input_token.token,
+                        Some(input_token.amount),
+                        output_token,
+                        None,
+                    ),
+                    QuoteTokens::OutputTokenSpecified {
+                        input_token,
+                        output_token,
+                        ..
+                    } => (
+                        swap_data.amount_input,
+                        input_token,
+                        None,
+                        output_token.token,
+                        Some(output_token.amount),
+                    ),
+                };
+                if let Some(expected_input_amount) = expected_input_amount {
+                    if expected_input_amount != swap_data.amount_input {
+                        return Err(RestError::BadParameters(
+                            format!(
+                                "Input amount in swap opportunity {} does not match the input amount in the swap instruction {}",
```

Comment:
nit but i think you can make the error messages a bit clearer to show that the searcher messed something up. Something like "Input amount in the submitted instruction is incorrect, does not match the input amount in the swap opportunity"